frontend/src/pages/casedesigner/detail/BookcaseDetail.tsx

```diff
@@ -75,32 +79,36 @@ export default function BookcaseDetail() {
   // Call to delete the bookcase
   const deleteBookcaseFinal = () => {
     logger.debug("Delete Bookcase Finalized");
-    setDeletePopupVisible(false);
+    setIsDeletePopupVisible(false);
     CASE_DESIGNER_API.deleteBookcase(id!)
       .then(() => {
         showSuccess(toast, "Bookcase deleted");
-        navigate("/bookcase");
+        navigate("/bookcases");
       })
       .catch(() => showFailure(toast, "Failed to delete bookcase"));
   };
 
-  const onSubmit = (): void => {
-    if (isAddPage) {
-      callAddBookcaseAPI();
-    } else {
-      callModifyBookcaseAPI();
-    }
-  };
-
   const callAddBookcaseAPI = () => {
     const APIbookcase = InternalToAPIBookcaseConversion(bookcase);
+    APIbookcase.name = bookcaseNameInSaveAsPopup;
     CASE_DESIGNER_API.addBookcase(APIbookcase)
-      .then(() => {
+      .then((response) => {
         showSuccess(toast, "Bookcase added successfully");
-        isGoBackActive ? navigate("/bookcases") : undefined;
+        setIsModifiable(!isModifiable);
+        setBookcase((draft) => {
+          draft.name = bookcaseNameInSaveAsPopup;
+        });
+        setOriginalData(APIToInternalBookcaseConversion(response));
       })
       .catch((error) => {
-        showFailure(toast, error.data.errors[0] ?? "Failed to add bookcase");
+        console.log(error);
+        const bookcaseNotUnique = error.data.errors?.name?.[0];
+        showFailure(
+          toast,
+          bookcaseNotUnique
+            ? "Bookcase not saved, name is not unique"
```

Comment:
I think this is possibly because you're not using the updated backend code. I made some changes to the backend and deployed locally, but unless you also did that you would be unable to use those changes